service/src/main/kotlin/io/provenance/digitalcurrency/consortium/domain/TxStatus.kt

```diff
@@ -40,9 +41,14 @@ open class TxStatusEntityClass(txEventTable: TxStatusTable) : UUIDEntityClass<Tx
 
     fun findForUpdate(uuid: UUID) = find { TST.id eq uuid }.forUpdate()
 
+    // TODO this should go awway after marker transfers are handled in batch
     fun findByTxHash(txHash: String) = find { TST.txHash eq txHash }
 
-    fun findByTxRequestUuid(txRequestUuid: UUID): List<TxStatusRecord> = find { TST.txRequestUuid eq txRequestUuid }.toList()
+    fun findAllByTxHash(txHash: String): SizedIterable<TxStatusRecord> =
+        find { TST.txHash eq txHash }
+
+    fun findByTxRequestUuid(txRequestUuid: UUID): List<TxStatusRecord> =
+        find { TST.txRequestUuid eq txRequestUuid }.toList()
```

Comment:
dumb question why some return list and others the sized iterable?

Reply:
it was my hacky way of trying to get around doing duplicated something like this:

```
val txStatusRecords: List<TxStatusRecord> = transaction { TxStatusRecord.findAllByTxHashForUpdate(txHash) }
            if (txStatusRecords.isEmpty()) {
                if (event is Migration && transaction { MigrationRecord.findByTxHash(txHash) == null }) {
                    handleMigrationEvent(txHash, event)
                } else if (event is Transfer &&
                    event.recipient == pbcService.managerAddress &&
                    event.denom == serviceProperties.dccDenom &&
                    transaction { MarkerTransferRecord.findByTxHash(txHash) == null }
                ) {
                    handleTransferEvent(txHash, event)
                }
            } else {
                transaction {
                    handleAllOtherEvents(txHash, TxStatusRecord.findAllByTxHashForUpdate(txHash))
                }
            }
```

I was wanting to keep transaction scope small because of the migration and marker transfer handlers.